src/pynwb/base.py

```diff
@@ -287,12 +286,124 @@ def __init__(self, **kwargs):
         self.images = images
 
 
-class TimeSeriesReference(namedtuple('TimeSeriesReference', ['idx_start', 'count', 'timeseries'])):
+class TimeSeriesReference(NamedTuple):
     """
     Class used to represent data values of a :py:class:`~pynwb.base.TimeSeriesReferenceVectorData`
-    This is a ``namedtuple('TimeSeriesReference', ['idx_start', 'count', 'timeseries'])`` type.
+    This is a ``typing.NamedTuple`` type with predefined tuple components
+    :py:meth:`~pynwb.base.TimeSeriesReference.idx_start`, :py:meth:`~pynwb.base.TimeSeriesReference.count`, and
+    :py:meth:`~pynwb.base.TimeSeriesReference.timeseries`.
+
+    :cvar idx_start:
+    :cvar count:
+    :cvar timeseries:
     """
-    pass
+    idx_start: int
+    """Start index in time for the timeseries"""
+
+    count: int
+    """Number of timesteps to be selected starting from :py:meth:`~pynwb.base.TimeSeriesReference.idx_start`"""
+
+    timeseries: TimeSeries
+    """The :py:class:`~pynwb.base.TimeSeries` object the TimeSeriesReference applies to"""
+
+    def check_types(self):
+        """
+        Helper function to check correct types for :py:meth:`~pynwb.base.TimeSeriesReference.idx_start`,
+        :py:meth:`~pynwb.base.TimeSeriesReference.count`, and :py:meth:`~pynwb.base.TimeSeriesReference.timeseries`.
+
+        This function is usually used in the try/except block to check for `TypeError` raised by the function.
+
+        See also :py:meth:`~pynwb.base.TimeSeriesReference.isvalid` to check both validity of types and the
+        reference itself.
+
+        :returns: True if successful. If unsuccessful `TypeError` is raised.
+
+        :raises TypeError: If one of the fields does not match the expected type
+        """
+        if not isinstance(self.idx_start, (int, np.integer)):
+            raise TypeError("idx_start must be an integer not %s" % str(type(self.idx_start)))
+        if not isinstance(self.count, (int, np.integer)):
+            raise TypeError("count must be an integer %s" % str(type(self.count)))
+        if not isinstance(self.timeseries, TimeSeries):
+            raise TypeError("timeseries must of of type TimeSeries. %s" % str(type(self.timeseries)))
```

Comment:
```suggestion
            raise TypeError("timeseries must of type TimeSeries. %s" % str(type(self.timeseries)))
```

Reply:
Will be resolved in upcoming commit since I also need to update tests to match.